### PR TITLE
Bootstrap for Ranger2 Admin

### PR DESCRIPTION
**Pull request by pritishsahoo**

### CODE REVIEW

File: hive/ranger2-client.sh [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+            esac
+        done
+        
+        URL=http://${HOST}:${PORT}
+        HIVE_LIB=/usr/lib/hive1.2
+	    
+	    S3_PATH=s3://paid-qubole/ranger-2.0.0
+	    RANGER_PLUGIN=ranger-2.0.0-hive-plugin
+    	RANGER_PLUGIN_FILE=${S3_PATH}/${RANGER_PLUGIN}.tar.gz
+  		SPOOL_DIR=/media/ephemeral0/logs/ranger/hive/audit/solr
```
Comment (himanshu-mishra) [inline hive/ranger2-client.sh]: Do we need to create this directory in this method itself or does Hive/Ranger take care of creating it ? Same with permission of directory.

File: hive/ranger2-client.sh [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+                    echo "Option -${OPTARG} requires an argument." >&2
+                    return 1
+                    ;;
+            esac
+        done
+        
+        URL=http://${HOST}:${PORT}
+        HIVE_LIB=/usr/lib/hive1.2
+	    
+	    S3_PATH=s3://paid-qubole/ranger-2.0.0
```
Comment (himanshu-mishra) [inline hive/ranger2-client.sh]: nit: Mix of spaces and tabs, let's replace tabs by spaces for consistency.

File: hive/ranger2-client.sh [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+    populate_nodeinfo
+    if [[ is_hs2_configured ]]; then
+        HOST=localhost
+        PORT=6080
+        REPOSITORY_NAME=hivedev
+        QBOL_HEALTH_CHECK_USER=qbol_user
+        SOLR_HOST=""
+        SOLR_PORT=6083
+        
+        while getopts ":h:p:r:S:P" opt; do
```
Comment (himanshu-mishra) [inline hive/ranger2-client.sh]: colon (:) missing after P

File: hive/ranger2-client.sh (hunk 1 above)
Comment (pritishsahoo) [reply hive/ranger2-client.sh]: Ranger will create it.

File: hive/ranger2-client.sh (hunk 2 above)
Comment (pritishsahoo) [reply hive/ranger2-client.sh]: Done

File: hive/ranger2-client.sh (hunk 3 above)
Comment (pritishsahoo) [reply hive/ranger2-client.sh]: Fixed.

**Comment by pritishsahoo** [comment]
Added Minor Changes to the Script.

**Review by himanshu-mishra — APPROVED** [review]
LGTM

File: hive/ranger-client.sh [hunk 4]
```diff
@@ -49,26 +67,46 @@ function install_ranger() {
 
         URL=http://${HOST}:${PORT}
         HIVE_LIB=/usr/lib/hive1.2
+
+        S3_PATH=s3://paid-qubole/ranger-${PLUGIN_VER}
+        RANGER_PLUGIN=ranger-${PLUGIN_VER}-hive-plugin
+        RANGER_PLUGIN_FILE=${S3_PATH}/${RANGER_PLUGIN}.tar.gz
+        SPOOL_DIR=/media/ephemeral0/logs/ranger/hive/audit/solr
```
Comment (kapils-qubole) [inline hive/ranger-client.sh]: nit: Move this to [here](https://github.com/qubole/bootstrap-functions/pull/35/files#diff-3b67c313e5a3b52155adacfce2d1980eR99) as it is only used in `if` block

**Review by kapils-qubole — APPROVED** [review]
Apart from a nit comment LGTM

**Comment by pritishsahoo** [comment]
Changes done.

**Review by akaranjkar-qu — CHANGES_REQUESTED** [review]
LGTM but need to add tests to [ranger-client.bats](https://github.com/qubole/bootstrap-functions/blob/master/tests/ranger-client.bats). Let me know if it is being handled in another JIRA

**Review by akaranjkar-qu — APPROVED** [review]

**Pull request merged**